presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -471,14 +471,46 @@ public void testSchemaAuthorizationForUser()
                 .setIdentity(Identity.forUser("user").withPrincipal(getSession().getIdentity().getPrincipal()).build())
                 .build();
 
+        Session anotherUser = testSessionBuilder()
+                .setCatalog(getSession().getCatalog().get())
+                .setSchema("test_schema_authorization_user")
+                .setIdentity(Identity.forUser("anotheruser").withPrincipal(getSession().getIdentity().getPrincipal()).build())
+                .build();
+
         // ordinary users cannot drop a schema or create a table in a schema the do not own
         assertQueryFails(user, "DROP SCHEMA test_schema_authorization_user", "Access Denied: Cannot drop schema test_schema_authorization_user");
         assertQueryFails(user, "CREATE TABLE test_schema_authorization_user.test (x bigint)", "Access Denied: Cannot create table test_schema_authorization_user.test");
 
         // change owner to user
         assertUpdate(admin, "ALTER SCHEMA test_schema_authorization_user SET AUTHORIZATION user");
 
+        // another user still cannot create tables
+        assertQueryFails(anotherUser, "CREATE TABLE test_schema_authorization_user.test (x bigint)", "Access Denied: Cannot create table test_schema_authorization_user.test");
+
         assertUpdate(user, "CREATE TABLE test_schema_authorization_user.test (x bigint)");
+        String selectSQL = "SELECT 1 FROM test_schema_authorization_user.test";
+        // TODO: This cannot find the schema for some reason
+        // assertQuery(user, selectSQL);
```

Comment:
@kokosing Any idea on this one? Every SELECT query fails with an error indicating that schema test_schema_authorization_user does not exist, even though I was just able to create a table in it.
It does not matter on behalf of which user I do this.

And it happens only in this test, not in a real cluster.

Any ideas?

Reply:
This is the exception: (I tried 1 and count(1), same result)
```
[ERROR] Tests run: 141, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 193.997 s <<< FAILURE! - in io.prestosql.plugin.hive.TestHiveIntegrationSmokeTest
[ERROR] testSchemaAuthorizationForUser(io.prestosql.plugin.hive.TestHiveIntegrationSmokeTest)  Time elapsed: 0.064 s  <<< FAILURE!
java.lang.AssertionError: Execution of 'expected' query failed: SELECT count(1) FROM test_schema_authorization_user.test
        at org.testng.Assert.fail(Assert.java:83)
        at io.prestosql.testing.QueryAssertions.assertQuery(QueryAssertions.java:161)
        at io.prestosql.testing.QueryAssertions.assertQuery(QueryAssertions.java:103)
        at io.prestosql.testing.AbstractTestQueryFramework.assertQuery(AbstractTestQueryFramework.java:125)
        at io.prestosql.plugin.hive.TestHiveIntegrationSmokeTest.testSchemaAuthorizationForUser(TestHiveIntegrationSmokeTest.java:493)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
        at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
        at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
        at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
        at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
        at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: org.jdbi.v3.core.statement.UnableToCreateStatementException: org.h2.jdbc.JdbcSQLSyntaxErrorException: Schema "TEST_SCHEMA_AUTHORIZATION_USER" not found; SQL statement:
SELECT count(1) FROM test_schema_authorization_user.test [90079-200] [statement:"SELECT count(1) FROM test_schema_authorization_user.test", rewritten:"SELECT count(1) FROM test_schema_authorization_user.test", parsed:"ParsedSql{sql='SELECT count(1) FROM test_schema_authorization_user.test', parameters=ParsedParameters{positional=false, parameterNames=[]}}", arguments:{positional:{}, named:{}, finder:[]}]
        at org.jdbi.v3.core.statement.SqlStatement.internalExecute(SqlStatement.java:1369)
        at org.jdbi.v3.core.result.ResultProducers.lambda$getResultSet$2(ResultProducers.java:68)
        at org.jdbi.v3.core.result.ResultIterable.lambda$of$0(ResultIterable.java:53)
        at org.jdbi.v3.core.result.ResultIterable.stream(ResultIterable.java:141)
        at org.jdbi.v3.core.result.ResultIterable.collect(ResultIterable.java:197)
        at org.jdbi.v3.core.result.ResultIterable.list(ResultIterable.java:186)
        at io.prestosql.testing.H2QueryRunner.execute(H2QueryRunner.java:185)
        at io.prestosql.testing.QueryAssertions.assertQuery(QueryAssertions.java:158)
        ... 16 more
Caused by: org.h2.jdbc.JdbcSQLSyntaxErrorException: Schema "TEST_SCHEMA_AUTHORIZATION_USER" not found; SQL statement:
SELECT count(1) FROM test_schema_authorization_user.test [90079-200]
        at org.h2.message.DbException.getJdbcSQLException(DbException.java:576)
        at org.h2.message.DbException.getJdbcSQLException(DbException.java:429)
        at org.h2.message.DbException.get(DbException.java:205)
        at org.h2.message.DbException.get(DbException.java:181)
        at org.h2.command.Parser.readTableFilter(Parser.java:1940)
        at org.h2.command.Parser.parseSelectFromPart(Parser.java:2827)
        at org.h2.command.Parser.parseSelect(Parser.java:2959)
        at org.h2.command.Parser.parseQuerySub(Parser.java:2817)
        at org.h2.command.Parser.parseSelectUnion(Parser.java:2649)
        at org.h2.command.Parser.parseQuery(Parser.java:2620)
        at org.h2.command.Parser.parsePrepared(Parser.java:868)
        at org.h2.command.Parser.parse(Parser.java:843)
        at org.h2.command.Parser.parse(Parser.java:815)
        at org.h2.command.Parser.prepareCommand(Parser.java:738)
        at org.h2.engine.Session.prepareLocal(Session.java:657)
        at org.h2.engine.Session.prepareCommand(Session.java:595)
        at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1235)
        at org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:76)
        at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:734)
        at org.jdbi.v3.core.statement.DefaultStatementBuilder.create(DefaultStatementBuilder.java:57)
        at org.jdbi.v3.core.statement.SqlStatement.internalExecute(SqlStatement.java:1361)
        ... 23 more
```

Reply:
BTW. I tried without any authorization changes and just with an admin user, and get the same result. Perhaps a bug in the JDBC code. I'll investigate more, but if you have an idea I'm eager to hear it.